main_test.go

```diff
@@ -734,36 +838,48 @@ func TestRotate(t *testing.T) {
 					log.Printf("Server stopped: %s", server.ListenAndServe())
 				}()
 				input := &Input{
-					SheetName:                    sheetNameMACFin,
 					UsernameHeader:               headingMACFinUsername,
 					PasswordHeader:               headinggMACFinPassword,
 					Bucket:                       "macfin",
 					Key:                          "pre1/pre2/macfin-dev-lbk-s3.xlsx",
 					AutomatedSheetPassword:       "asfas",
-					AutomatedSheetName:           "PasswordManager",
 					AutomatedSheetColNameToIndex: cols,
 					RowOffset:                    1,
+					SheetGroups: map[Environment]SheetGroup{
+						dev: {
+							AutomatedSheetName: "PasswordManager",
+							SheetName:          sheetNameMACFin,
+						},
+					},
 				}
 
-				portal := &Portal{
-					Hostname:    portalServer,
-					IDMHostname: idmServer,
-					Scheme:      "http://",
+				envToPortal := map[Environment]*Portal{
+					dev: {
+						Hostname:    portalServer,
+						IDMHostname: idmServer,
+						Scheme:      "http://",
+					},
 				}
 
 				fc := &FakeS3Client{
 					Bucket:                       input.Bucket,
 					Key:                          input.Key,
 					LocalPath:                    filename,
-					AutomatedSheetName:           input.AutomatedSheetName,
+					AutomatedSheetName:           input.SheetGroups[dev].AutomatedSheetName,
 					AutomatedSheetColNameToIndex: input.AutomatedSheetColNameToIndex,
 					RowOffset:                    input.RowOffset,
 				}
-				err = rotate(input, portal, fc)
+				err = rotate(input, envToPortal, fc)
 				server.Shutdown(context.Background())
 
 				if err != nil {
-					t.Fatalf("Error running rotate(): %s", err)
+					if errors.Is(err, ErrSheetDoesNotExist) || errors.Is(err, ErrSheetIsEmpty) || errors.Is(err, ErrSheetMissingHeader) ||
+						errors.Is(err, ErrWrongNumberOfCols) || errors.Is(err, ErrWrongColumnHeading) {
+						t.Logf("Error in input file: %s", err)
+						return
+					} else {
+						t.Fatalf("Error running rotate(): %s", err)
+					}
```

Comment:
you also need to fail the test case if there is _no_ error but there was an invalid input.

Reply:
yup!